File: Day346/LargestSubStringBetweenTwoCharacters.java
class Solution {
    public int maxLengthBetweenEqualCharacters(String s) {
        Map<Character, Integer> map = new HashMap<>();
        int answer = 0;
        int i = 0;
        for (char c: s.toCharArray()) {
            if (map.containsKey(c)) {
                answer = Math.max(answer,i - map.get(c));
            } else {
                map.put(c, i);
            }
            i++;
        }
        return answer-1;
    }
}

File: Day346/LargestSubStringBetweenTwoCharacters.go
func maxLengthBetweenEqualCharacters(s string) int {
    m := make(map[rune]int)
    answer := 0
    for i,v := range s {
        if _,flag:= m[v]; flag {
            answer = int(math.Max(float64(answer), float64(i - m[v])))
        } else {
            m[v] = i
        }
    }
    return answer-1
}